object/merge.test.ts

```diff
@@ -0,0 +1,121 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import merge from "./merge.ts";
+
+describe("merge", () => {
+  it("should merge with empty object", () => {
+    const a: object = {};
+    const b: object = { a: 1 };
+
+    const received = merge(a, b);
+    const expected: object = { a: 1 };
+
+    expect(received).toEqual(expected);
+  });
+
+  it("should merge and override the same property", () => {
+    const a: object = { a: 1, b: 3 };
+    const b: object = { b: 7 };
+
+    const received = merge(a, b);
+    const expected: object = { a: 1, b: 7 };
+
+    expect(received).toEqual(expected);
+  });
+
+  it("should merge properties which are objects", () => {
+    const a: object = { a: 1, b: { c: 3 } };
+    const b: object = { b: { d: 8 } };
+
+    const received = merge(a, b);
+    const expected: object = { a: 1, b: { c: 3, d: 8 } };
+
+    expect(received).toEqual(expected);
+  });
+
+  it("should merge two objects with different types of properties", () => {
```

Comment:
Consider rephrasing this to "it handles only object-merges but keeps non-conflicting values intact".